packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt

```diff
@@ -155,6 +163,13 @@ public interface AppConfiguration {
             this.syncRootDirectory = rootDir
         }
 
+        /**
+         * TODO
+         */
+        public fun defaultSyncClientResetStrategy(
+            strategy: SyncClientResetStrategy
```

Comment:
This allows users to create new types of strategies like:

defaultSyncClientResetStrategy(object: SyncClientResetStrategy {
                    
                })

We should have functions for our supported ones, discard and local.